### DISCUSSION TITLE
How can I solve auto parameter not permitted in this context error, to install Inviwo ?

### DISCUSSION THREAD

**Discussion by faltefalte** [inviwo/inviwo#1282] [Q&A]
I have a compilation problem to install Inviwo (a software). I've got this following error :

```
error: ‘auto’ parameter not permitted in this context
util::Combinations comb(util::span{bases}, searchGroupSize);
```

Do you know where it can come from and how to solve it please? I'm on Linux.

Thank you

**Comment by petersteneteg** [comment]
Hi 
Do you know what compiler and version you are using?
Inviwo require fairly good C++17 support. 
Might also be useful if you can provide the full log.

**Reply by faltefalte** [reply]
It works with C++17, thank you !

**Comment by ResearchDaniel** [comment]
We get the same error in the CI system:
[114/1373] Building CXX object modules/core/CMakeFiles/inviwo-core.dir/datastructures/unitsystem.cpp.o
FAILED: modules/core/CMakeFiles/inviwo-core.dir/datastructures/unitsystem.cpp.o 
/usr/bin/c++ -DCMAKE_BUILD_TYPE=\"Release\" -DCMAKE_CXX_COMPILER_ID=\"GNU\" -DCMAKE_CXX_COMPILER_VERSION=\"9.4.0\" -DCMAKE_GENERATOR=\"Ninja\" -DENABLE_UNIT_MAP_ACCESS -DFMT_LOCALE -DGLM_ENABLE_EXPERIMENTAL -DGLM_FORCE_SILENT_WARNINGS -DGLM_FORCE_SIZE_T_LENGTH -DGLM_FORCE_UNRESTRICTED_GENTYPE -DHAVE_CONFIG_H -DINVIWO_ALL_DYN_LINK -DINVIWO_CORE_EXPORTS -DIVW_FORCE_ASSERTIONS -DIVW_PROFILING -DIVW_RELEASE -DTCB_SPAN_NAMESPACE_NAME=inviwo::util -DUNITS_EXPORT_HEADER -I/home/runner/work/inviwo/inviwo/build/modules/core/include -I/home/runner/work/inviwo/inviwo/inviwo/include -I/home/runner/work/inviwo/inviwo/inviwo/ext/ticpp/include -I/home/runner/work/inviwo/inviwo/inviwo/ext/tclap/include -I/home/runner/work/inviwo/inviwo/inviwo/ext/warn/include -I/home/runner/work/inviwo/inviwo/inviwo/ext/sml/include -I/home/runner/work/inviwo/inviwo/inviwo/ext/span/span/include -I/home/runner/work/inviwo/inviwo/inviwo/ext/half/include -I/home/runner/work/inviwo/inviwo/inviwo/ext/flags/include -I/home/runner/work/inviwo/inviwo/inviwo/tools/tracy/include -I/home/runner/work/inviwo/inviwo/inviwo/ext/units/units -I/home/runner/work/inviwo/inviwo/build/ext/units/units/units -I/home/runner/work/inviwo/inviwo/build/ext/units/units -I/home/runner/work/inviwo/inviwo/inviwo/ext/CRoaring/CRoaring/include -I/home/runner/work/inviwo/inviwo/inviwo/ext/CRoaring/CRoaring/include/roaring -I/home/runner/work/inviwo/inviwo/inviwo/ext/CRoaring/CRoaring/cpp -isystem /home/runner/work/inviwo/inviwo/build/vcpkg_installed/x64-linux/include -O3 -DNDEBUG -fPIC -Wall -Wextra -pedantic -Wno-unused-parameter -Wno-missing-braces -std=c++17 -MD -MT modules/core/CMakeFiles/inviwo-core.dir/datastructures/unitsystem.cpp.o -MF modules/core/CMakeFiles/inviwo-core.dir/datastructures/unitsystem.cpp.o.d -o modules/core/CMakeFiles/inviwo-core.dir/datastructures/unitsystem.cpp.o -c /home/runner/work/inviwo/inviwo/inviwo/src/core/datastructures/unitsystem.cpp
/home/runner/work/inviwo/inviwo/inviwo/src/core/datastructures/unitsystem.cpp: In function ‘std::pair<double, std::vector<std::tuple<std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, int> > > inviwo::util::findBestSetOfNamedUnits(inviwo::Unit, const EnabledGroups&, inviwo::util::UseUnitPrefixes)’:
Error: /home/runner/work/inviwo/inviwo/inviwo/src/core/datastructures/unitsystem.cpp:163:43: error: ‘auto’ parameter not permitted in this context
  163 |             util::Combinations comb(util::span{bases}, searchGroupSize);
      |

**Reply by petersteneteg** [reply]
The compiler version is too old, try a newer version of gcc
